src/methods/method_overloading/hard/q6/MainTest.java

```diff
@@ -0,0 +1,131 @@
+package methods.method_overloading.hard.q6;
+
+import java.util.stream.Stream;
+
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.PlaceHolder;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "Jack Swift\nJack.Swift\n38471";
+        return new Clause[]{
+                new StringLiteral("Please enter a student's name:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("Please enter a username:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("Please enter a student number:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new PlaceHolder(),
+                new StringLiteral(" is a valid username: "),
+                new PlaceHolder(),
+                new NewLine(),
+                new IntegerLiteral("studentNum"),
+                new StringLiteral(" is a valid student number: "),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        AttendingStudent.main(new String[0]);
+    }
+
+    static Stream<Arguments> studentNumberProvider() {
+        return Stream.of(
+                Arguments.of(12345678, true),
```

Comment:
In the case they check the length using numbers instead of a string, you should add a case for 100000000 (which would be the first number above eight digits)

Reply:
aha whoops, looks like when I was rewriting the question I messed up the question a bit, it's supposed to say at least 8 digits not equal to 8 digits

Reply:
Ah all good then